### DISCUSSION TITLE
(manager/gradle) Support properties map in gradle kts

### DISCUSSION THREAD

**Discussion by guicamest** [renovatebot/renovate#21425] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Similar to https://github.com/renovatebot/renovate/pull/18990, the `properties` map can also be supported. *Note* the `properties[]` call instead of `property()`.

Given a `gradle.properties` file that declares a version for a dependency:
```
...
mockitoKotlin=4.0.0
```
and a build.gradle.kts that declares a dependency that uses that property via the `properties` map:
```
dependencies {
   ...
   testImplementation("org.mockito.kotlin:mockito-kotlin:${properties["mockitoKotlin"]}")
   ...
}
```
I would like renovate to raise a PR to update mockitoKotlin to the latest version (current is 4.1.0)

### What you tried so far.

I tried of adding an extra `properties` value to the `tokenValues` list in [parser/common.ts](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/gradle/parser/common.ts#L94) and added a test which is currently passing.

Given the implementation of https://github.com/renovatebot/renovate/pull/18990 is somewhat different, wanted to double check before submitting a PR.

**Comment by Churro** [comment, marked as answer]
Makes sense to me. PR #18990 has added initial support for properties and is therefore a bit larger. 

Would be great if you could submit a PR to add this pattern (Gradle ref available [here](https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties)) 👍

**Reply by guicamest** [reply]
I'll get to it then 👍
